Display userinfo (#484)
kyma-project/luigi: docs/authorization-configuration.md

    scope: 'audience:server:client_id:client openid profile email groups',
    redirect_uri: '',
    post_logout_redirect_uri: '/logout.html',
    automaticSilentRenew: true,

>     userInfoFn:()=>{}

Reviewer: As a user, I'd like to get token info here, same the provider receives. `const tokenInfo = this.parseIdToken(authData.idToken);`  But watch out, OIDC provider does not have `getAuthData` function yet.
Might be enough to hand over `JSON.parse(localStorage.getItem('luigi.auth'))`.

- **automaticSilentRenew** enables the automatic silent renewal of the token if it is supported by the server. The default value is `false`. For this mechanism to work, the browser must have third-party cookies support enabled.
- **accessTokenExpiringNotificationTime** is the number of seconds before an access token is to expire and triggers silent token refresh. The default value is 60.
- **thirdPartyCookiesScriptLocation** is the URL to the page containing third-party cookies support check. For details, see [Third-party cookies and silent token refresh section](#Third-party-cookies-and-silent-token-refresh).

> - **userInfoFn** provides a function to get user information. It returns a promise of a user info object which contains user name and/or email to display in the profile dropdown.

Reviewer 2: ```suggestion
- **userInfoFn** provides a function to get user information. It returns a promise of a **userinfo** object which contains user name and/or email to display in the profile dropdown menu.
```

- **response_type** defaults to the **id_token**. Any other parameter that is added to oAuthData is also added to the authorization payload.
- **nonceFn** provides a function that returns a string in order to override the default **nonce**.
- **logoutFn** provides the function to override the **logoutUrl** functionality for a custom logout. It needs to execute the **logoutCallback()** function after logout.

> - **userInfoFn** provides a function to get user information. It returns a promise of a user info object which contains user name and/or email to display in the profile dropdown.

Reviewer 2: ```suggestion
- **userInfoFn** provides a function to get user information. It returns a promise of a **userinfo** object which contains user name and/or email to display in the profile dropdown menu.
```

> ### Custom Authentification Provider

Reviewer 2: I'm wondering, should it be `authentification` or `authentication`? I was researching similar providers and it was `authentication` 🤔  but I might be wrong :) However, it is used to authenticate the user right?

Author: You are totally right :)

> If you are using any authentification provider you also have the possibility to implement some functions that luigi can deal with.

Reviewer 2: ```suggestion
If you are using any authentication provider you can also implement some other functions for Luigi to use.
```

````
export class CustomAuthentificationProvider {

    login(){
        // logic to handle the login mechanism

>         // returns a promise which contains a error message if something went wrong

Reviewer 2: ```suggestion
        // returns a promise which contains an error message if something went wrong
```

    }

    logout(authData, logoutCallback){
        // logic to handle the logout mechanism
    }

    setTokenExpirationAction(){
    }

    generateNonce(){
        //returns a string 
    }

    userInfo(){
        // logic to get some user information

>         // returns a promise of a userinfo object which contains user name and/or email to display in the profile dropdown menu

Reviewer 2: ```suggestion
        // returns a promise of a userinfo object which contains a user name and/or email to display in the profile dropdown menu
```
